arch/risc-v/src/rv64gc/riscv_assert.c

```diff
@@ -255,18 +258,48 @@ static void up_dumpstate(void)
   _alert("stack size: %016" PRIxPTR "\n", ustacksize);
 #endif
 
+#ifdef CONFIG_ARCH_KERNEL_STACK
+  /* Does this thread have a kernel stack allocated? */
+
+  if (rtcb->xcp.kstack)
+    {
+      kstackbase = (uintptr_t)rtcb->xcp.kstack;
+
+      _alert("Kernel stack:\n");
+      _alert("  base: %016\n" PRIxPTR "\n", kstackbase);
+      _alert("  size: %016\n" PRIxPTR "\n", CONFIG_ARCH_KERNEL_STACKSIZE);
+    }
+#endif
+
   /* Dump the user stack if the stack pointer lies within the allocated user
    * stack memory.
    */
 
   if (sp >= ustackbase && sp < ustackbase + ustacksize)
     {
-      _alert("ERROR: Stack pointer is not within allocated stack\n");
+      _alert("User Stack\n");
       up_stackdump(ustackbase, ustackbase + ustacksize);
     }
+
+#ifdef CONFIG_ARCH_KERNEL_STACK
+  /* Dump the kernel stack if the stack pointer lies within the allocated
+   * kernel stack memory.
+   */
+
+  else if (sp >= kstackbase &&
+           sp < kstackbase + CONFIG_ARCH_KERNEL_STACKSIZE)
+    {
+      _alert("Kernel Stack\n");
+      up_stackdump(sp, kstackbase + CONFIG_ARCH_KERNEL_STACKSIZE);
+    }
+#endif
   else
     {
+      _alert("ERROR: Stack pointer is not within allocated stack\n");
```

Comment:
maybe we should know here whether we *should* be in kernel or user stack. And also check if sp is < stack base or > stack base+stack size.